platform/node/index.d.ts

```diff
@@ -0,0 +1,49 @@
+declare module '@maplibre/maplibre-gl-native' {
+  enum Resource {
+    Unknown = 0,
+    Style = 1,
+    Source = 2,
+    Tile = 3,
+    Glyphs = 4,
+    SpriteImage = 5,
+    SpriteJSON = 6,
+  }
+
+  type RequestResponse = {
+    data: ArrayBuffer;
+    modified?: Date;
+    expires?: Date;
+    etag?: string;
+  };
+
+  type MapOptions = {
+    request: (
+      request: { url: string; kind: Resource },
+      callback: (error?: Error, response?: RequestResponse) => void,
+    ) => void;
+    ratio?: number;
+  };
+
+  type RenderOptions = {
+    zoom?: number; // defaults to 0
```

Comment:
Please use jsdoc style comments, and also add comments to the classes and other types.

Reply:
Thanks for this proposal. Would be awesome if you could have another look!